mackerel_test.go

```diff
@@ -34,6 +34,48 @@ func TestRequest(t *testing.T) {
 	}
 }
 
+func Test_requestInternal(t *testing.T) {
+	tests := []struct {
+		method               string
+		body                 io.Reader
+		hasContentTypeHeader bool
+	}{
+		{"GET", nil, false},
+		{"POST", nil, true},
+		{"PUT", nil, true},
+		{"DELETE", nil, true},
+		{"GET", strings.NewReader("some"), true},
+		{"POST", strings.NewReader("some"), true},
+		{"PUT", strings.NewReader("some"), true},
+		{"DELETE", strings.NewReader("some"), true},
+	}
+	for _, test := range tests {
+		t.Run(fmt.Sprintf("%s with %v body", test.method, test.body), func(tt *testing.T) {
+			ts := httptest.NewServer(http.HandlerFunc(func(res http.ResponseWriter, req *http.Request) {
+				if !test.hasContentTypeHeader && req.Header.Get("Content-Type") == "application/json" {
+					t.Error("Content-Type header should not have application/json")
+				}
+				if test.hasContentTypeHeader && req.Header.Get("Content-Type") != "application/json" {
+					t.Error("Content-Type header should have application/json")
+				}
+				res.Write([]byte(`{"success": true}`)) // nolint
+			}))
```

Comment:
How about a comment that this test server's behavior is similar to Mackerel's public API?